### ISSUE TITLE
German Umlaut in Maven stdout causes SAM build to fail

### ISSUE THREAD

**Issue by InsertPikachuFace** [aws/aws-lambda-builders#548]
### Description:

Because the log output from the Maven build is decoded with UTF-8 ([see line](https://github.com/aws/aws-lambda-builders/blob/a01d058be95a64b8ffdb80ee51c7edaf783686b9/aws_lambda_builders/workflows/java_maven/maven.py#L31)), an "ü" in our unit-test logs causes the SAM local Lambda build to fail.
If I add -DskipTests with a Makefile it completes as expected.

### Steps to reproduce the issue:

1. Set up Lambda project that logs out an Umlaut (äÄöÖüÜß) during `maven package`
2. run `sam build` with default template configuration

### Observed result:
```
 | JavaMavenWorkflow:CopySource succeeded
 | Running JavaMavenWorkflow:MavenBuild
 | JavaMavenWorkflow:MavenBuild raised unhandled exception
Traceback (most recent call last):
  File "C:\Program Files\Amazon\AWSSAMCLI\runtime\lib\site-packages\aws_lambda_builders\workflow.py", line 369, in run
    action.execute()
  File "C:\Program Files\Amazon\AWSSAMCLI\runtime\lib\site-packages\aws_lambda_builders\workflows\java_maven\actions.py", line 39, in execute
    self.subprocess_maven.build(self.scratch_dir)
  File "C:\Program Files\Amazon\AWSSAMCLI\runtime\lib\site-packages\aws_lambda_builders\workflows\java_maven\maven.py", line 31, in build
    LOG.debug("Maven logs: %s", stdout.decode("utf8").strip())
UnicodeDecodeError: 'utf-8' codec can't decode byte 0xfc in position 43840: invalid start byte
```

### Expected result:
SAM build completes despite an encoding error in a debug-log or it logs out any Maven output correctly.

### Additional environment details (Ex: Windows, Mac, Amazon Linux etc)

1. OS: Windows 10
4. SAM CLI version: 1.96.0
5. AWS region: EU-CENTRAL-1

**Comment by jfuss** [comment]
Thanks for reporting. Going to mark this as a bug.

@InsertPikachuFace Is this something you would want to contribute a fix for? We understand if not, just will be some lead time before we can patch. We do prioritize bugs highly regardless just might be faster if you submitted a patch is all.

**Comment by jfuss** [comment]
@InsertPikachuFace No worries at all. We appreciate you taking the time to report.

**Comment by mildaniel** [comment]
The fix has been released in Lambda Builders version 1.40.0 and in AWS SAM CLI version 1.99.0.
